pkg/block/fetcher.go

```diff
@@ -706,10 +706,16 @@ func (r *ReplicaLabelRemover) Modify(_ context.Context, metas map[ulid.ULID]*met
 type ConsistencyDelayMetaFilter struct {
 	logger           log.Logger
 	consistencyDelay time.Duration
+	bkt              objstore.InstrumentedBucketReader
 }
 
 // NewConsistencyDelayMetaFilter creates ConsistencyDelayMetaFilter.
-func NewConsistencyDelayMetaFilter(logger log.Logger, consistencyDelay time.Duration, reg prometheus.Registerer) *ConsistencyDelayMetaFilter {
+func NewConsistencyDelayMetaFilter(
+	logger log.Logger,
+	consistencyDelay time.Duration,
+	bkt objstore.InstrumentedBucketReader,
```

Comment:
If we want we could just add interface for Attributes so the usability is scoped here (it's important to know if component has access to writing/reading blocks or not.

Reply:
I agree. Umm, I cannot name this interface :rofl:  `AttrReader`?